oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/MergeRunnerTest.java

```diff
@@ -0,0 +1,176 @@
+package org.apache.jackrabbit.oak.index.indexer.document.flatfile;
+
+import com.google.common.collect.Lists;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.event.Level;
+import org.apache.commons.io.FileUtils;
+import org.apache.jackrabbit.oak.commons.junit.LogCustomizer;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Phaser;
+
+import static com.google.common.base.Charsets.UTF_8;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class MergeRunnerTest {
+    private final LogCustomizer lc = LogCustomizer.forLogger(MergeRunner.class)
+            .filter(Level.INFO)
+            .enable(Level.INFO)
+            .create();
+    private final String newline = System.lineSeparator();
+    private final List<File> testFiles = Lists.newArrayList();
+    private String expectedSortedFileStr = "",
+                   actualSortedFileStr = "";
+
+    @Before
+    public void setup(){
+        lc.starting();
+    }
+
+    @After
+    public void after() {
+        lc.finished();
+    }
+
+    @Test
+    public void test() throws IOException {
+        lc.starting();
+        System.setProperty(FlatFileNodeStoreBuilder.PROP_MERGE_TASK_BATCH_SIZE, "3");
```

Comment:
If you are setting the system properties in the test, then you would need to use 

    try {
        old = System.getProperty...
        System.setProperty...
    } finally {
        System.setProperty old...
    }

But! It's better to use a setter, and not change the system properties (see above).

Reply:
make it part of the param